### PR TITLE
finished the task jv-oop-lesson-0 "Working with services"

### PR DESCRIPTION
**Pull request by ivankusur**

### CODE REVIEW

File: src/main/java/mate/academy/service/UserService.java [hunk 1]
```diff
@@ -3,18 +3,25 @@
...
     /**
      * Find user by email. All users are stored in <code>private static final User[] users</code>
+     *
      * @param email - the input parameter
      * @return - user if his email is equal to passed email.
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
-        return null;
+        User returnUser = new User("USER NOT CONTAIN", "USER NOT CONTAIN");
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/UserService.java:19]:
```suggestion
```

File: src/main/java/mate/academy/service/UserService.java [hunk 2]
```diff
@@ -3,18 +3,25 @@
...
+        for (User user : users) {
+            if (email.equals(user.getEmail())) {
+                returnUser = user;
+            }
+        }
+        return returnUser;
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/UserService.java:25]: Return <code>null</code> if there is no suitable user

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 3]
```diff
@@ -10,7 +12,13 @@ public class AuthenticationService {
      * @return true if user by email exists and passed password is equal to user's password.
      * Return false in any other cases.
      */
+    private UserService userService = new UserService();
+
     public boolean login(String email, String password) {
-        return false;
+        if (!Objects.equals(userService.findByEmail(email).getEmail(), email)
+                || !Objects.equals(userService.findByEmail(email).getPassword(), password)) {
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/AuthenticationService.java:19]: try to avoid using IF
do not call `findByEmail` 2 times

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/UserService.java (hunk 2 above)
Comment (ivankusur) [reply src/main/java/mate/academy/service/UserService.java:25]: but when I add NULL, tests fail

**Comment by ivankusur** [comment]
but when I add NULL, tests fail

**Comment by ivankusur** [comment]
Final implementation, fixed that tests NOT fail with NULL.

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 4]
```diff
@@ -10,7 +12,10 @@ public class AuthenticationService {
      * @return true if user by email exists and passed password is equal to user's password.
      * Return false in any other cases.
      */
+    private UserService userService = new UserService();
+
     public boolean login(String email, String password) {
-        return false;
+        User user = userService.findByEmail(email);
+        return user != null && user.getEmail().equals(email) && user.getPassword().equals(password);
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/AuthenticationService.java:19]:
```suggestion
        return user != null && user.getPassword().equals(password);
```

File: src/main/java/mate/academy/service/UserService.java [hunk 5]
```diff
@@ -3,18 +3,25 @@
...
      * @param email - the input parameter
      * @return - user if his email is equal to passed email.
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
-        return null;
+        User returnUser = null;
+        for (User user : users) {
+            if (email.equals(user.getEmail())) {
+                returnUser = user;
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/UserService.java:22]:
```suggestion
                return user;
```

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**